includes/data-stores/class-wc-admin-reports-coupons-data-store.php

```diff
@@ -249,4 +258,48 @@ protected function get_cache_key( $params ) {
 		return 'woocommerce_' . self::TABLE_NAME . '_' . md5( wp_json_encode( $params ) );
 	}
 
+	/**
+	 * Make an entry in the wc_order_coupon_lookup table for an order.
+	 *
+	 * @since 3.5.0
+	 * @param int $order_id Order ID.
+	 * @return void
+	 */
+	public static function sync_order_coupons( $order_id ) {
```

Comment:
What do you think about some unit tests for this method?

Reply:
Sounds like a smart move.  Just to check before implementing this, are we okay fixing up existing tests to check for overall coverage?

I believe most of this code is covered by the save triggered in `tests/reports/class-wc-tests-reports-coupons.php`, but we need to amend this with order status data to fully cover that side of things.

Or do you think we need to unit test specifically against this method?

Reply:
> I believe most of this code is covered by the save triggered in tests/reports/class-wc-tests-reports-coupons.php, but we need to amend this with order status data to fully cover that side of things.

Ah I was not aware of that - seems sufficient then.

Reply:
Opened an issue to increase coverage for those 2 points.  Probably needs to be done across all tests.  https://github.com/woocommerce/wc-admin/issues/1193